lib/on_form/form.rb

```diff
@@ -91,8 +91,13 @@ def self.expose_collection_of(association_name, on: nil, prefix: nil, suffix: ni
       collection_form_class.take_identity_from singular_name, convert_to_model: false
       collection_form_class.class_eval(&block)
 
-      define_method(exposed_name) { collection_wrappers[association_name] ||= CollectionWrapper.new(backing_model_instance(on), association_name, collection_form_class, allow_insert, allow_update, allow_destroy) } # used by action_view's fields_for, and by the following lines
+      define_method(exposed_name) do
+        collection_wrappers[association_name] ||= CollectionWrapper.new(backing_model_instance(on), association_name, collection_form_class, allow_insert, allow_update, allow_destroy) # used by action_view's fields_for, and by the following lines
+      end
+
       define_method("#{exposed_name}_attributes=") { |params| send(exposed_name).parse_collection_attributes(params) }
+      define_method("_validate_#{exposed_name}_forms") { send(exposed_name).validate_forms(self) }
```

Comment:
If we go with this approach, I'd rather do something like

```
send(exposed_name).validation_errors.each { |error| self.errors.add(exposed_name, error) }
```

rather than add more coupling between self and the collection wrapper

Reply:
Yep, we can do that. Another problem here adding child form validation and child form backing model validation adding here. Before adding parent model backing validation error. Its looking odd while displaying the error message.

I'm trying to solve this problem on https://github.com/fluxfederation/on_form/pull/6. 
ie: Always run all form validation first then go for all backing model validation.